dms/models/mixins_thumbnail.py

```diff
@@ -14,73 +14,53 @@ class Thumbnail(models.AbstractModel):
     _name = "dms.mixins.thumbnail"
     _description = "Thumbnail Mixin"
 
-    custom_thumbnail = fields.Binary(
-        string="Custom Thumbnail", attachment=False, prefetch=False,
-    )
-    custom_thumbnail_medium = fields.Binary(
-        "Medium Custom Thumbnail", compute="_compute_custom_thumbnail", prefetch=False,
-    )
-    custom_thumbnail_small = fields.Binary(
-        "Small Custom Thumbnail", compute="_compute_custom_thumbnail", prefetch=False,
-    )
-
-    thumbnail = fields.Binary(
-        compute="_compute_thumbnail", string="Thumbnail", prefetch=False,
-    )
+    custom_thumbnail = fields.Binary(string="Custom Thumbnail", attachment=False)
+    thumbnail = fields.Binary(compute="_compute_thumbnail", store=True)
```

Comment:
This change may be the hell on existing databases when updating the module. Why the `prefetch=False` is not working?

Reply:
[`prefetch=False` is the default for binary fields](https://github.com/odoo/odoo/blob/18c6e059180367998f1c46efa0dc4697df67f0f3/odoo/fields.py#L1852), so it's not needed to declare it.

Reply:
OK, but why is this not working then?

Reply:
I insist that this change is no good, and I prefer to not do it.

Reply:
The main change is to make it `store=True`. It is important to do it because thumbnails exist to speed up things, and having to go through batch image conversion on each request kills that purpose.

That's how odoo thumbnails work elsewhere AFAIK.

Reply:
This is still no good. These fields shouldn't be computed unless they are used, so the problem is there. Please don't store it.

Reply:
If I don't store them, then I have to remove them.

The only purpose for a thumbnail's existence is to generate a cache and speed things up, and that's why everywhere else Odoo stores computed thumbnails.

In fact, you can see that [in odoo 13.0 we have the `image.mixin` class that implements exactly this same behavior](https://github.com/odoo/odoo/blob/6a7db37d5cc6316cf51bf9b1eea5dcee0aebe97c/odoo/addons/base/models/image_mixin.py#L15-L19).

If Odoo needs to resize images on each request, things become slow. If there are 80 thumbnails in a screen, then it becomes a bottleneck.

So, I'll proceed to remove all these fields.

Reply:
No, don't remove them, but as said, if they are not used, they shouldn't be computed. Why are they computed? You are not answering my question since the beginning, and that's the key.

Reply:
Well, you didn't ask any questions. You just said "this is hell", "this is no good", etc. _That_ is the key.

Now regarding why they are computed: because they are used.

Reply:
I can stop using them instead of removing them, if you prefer. However they'll still be garbage anyway. 🤷🏼‍♂️

Reply:
You aren't still answering me where they are used.